src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs

```diff
@@ -2,37 +2,136 @@
+                    await InputSimulator.SendAsync(
+                        form,
+                        inputSimulator => inputSimulator.Keyboard
+                        .Sleep(100)
+                        .KeyPress(WindowsInput.Native.VirtualKeyCode.UP));
+
+                    await Task.Delay(100);
+
+                    stringBuilder.Append($"i:{i + 1}, eventsCount:{comboBox.EventsCount}\n");
+                }
+
+                TestOutputHelper.WriteLine($"{stringBuilder}");
+                Assert.Equal(keyPressesCount, comboBox.EventsCount);
+            },
+            numberOfItemsToAdd: 10,
+            selectedIndex: 9);
+        }
+
+        [WinFormsFact]
+        public async Task ComboBox_Select_Item_By_DownArrowKeyAsync()
+        {
+            await RunComboBoxTestAsync(async (form, comboBox) =>
+            {
+                // The EventsCount property needs to be reset because the conrol is being selected right after the creation.
+                comboBox.ResetCount();
+                int keyPressesCount = 9;
+                for (int i = 0; i < keyPressesCount; i++)
+                {
+                    await InputSimulator.SendAsync(
+                        form,
+                        inputSimulator => inputSimulator.Keyboard.KeyPress(WindowsInput.Native.VirtualKeyCode.DOWN));
+
+                    Assert.Equal(i + 1, comboBox.EventsCount);
+                }
+
+                Assert.Equal(keyPressesCount, comboBox.EventsCount);
+            },
+            numberOfItemsToAdd: 10,
+            selectedIndex: 0);
+        }
+
+        private async Task RunComboBoxTestAsync(Func<Form, ComboBoxWithSelectCounter, Task> runTest, int numberOfItemsToAdd, int selectedIndex)
+        {
+            await RunSingleControlTestAsync(
+                testDriverAsync: runTest,
+                createControl: () =>
+                {
+                    ComboBoxWithSelectCounter control = new();
+                    AddItems(control, numberOfItemsToAdd);
+                    control.SelectedIndex = selectedIndex;
+                    return control;
+                },
+                createForm: () =>
+                {
+                    return new()
+                    {
+                        Size = new(500, 300),
+                    };
+                });
+
+            static void AddItems(ComboBoxWithSelectCounter comboBox, int numToAdd)
+            {
+                for (int i = 0; i < numToAdd; i++)
+                {
+                    comboBox.Items.Add($"Item {i}");
+                }
+            }
+        }
+
+        private class ComboBoxWithSelectCounter : ComboBox
+        {
+            public int EventsCount { get; private set; }
+            protected override void OnSelectedIndexChanged(EventArgs e)
+            {
+                base.OnSelectedIndexChanged(e);
+                EventsCount++;
+            }
+
+            public void ResetCount()
```

Comment:
```suggestion
            public void ResetEventsCount()
```

...or we can leave `public int EventsCount { get; set; }`

Reply:
Let's stick with ResetEventsCount() than.